### PR TITLE
Simplified threading logic

### PR DESCRIPTION
**Pull request by ulteq**
The physics worker thread is now handled by the thread pool:
- Gets rid of the dummy thread pool which was used as a background worker thread
- Makes the `DisableThreadPool` setting obsolete
- Renamed `NumThreadsInThreadPool` into `NumWorkerThreads`
- Introduces `GVarPod_APS<int> app_num_workers`
- Converts `GVarPod_A<bool> app_multithread` into `GVarPod_APS<bool> app_async_physics`

- [x] Decide if we should rename `RoR::App::app_multithread` into `RoR::App::app_async_physics`

Reference: https://github.com/only-a-ptr/rigs-of-rods/pull/22

### CODE REVIEW

**Comment by only-a-ptr** [comment]
> Decide if we should rename RoR::App::app_multithread into RoR::App::app_async_physics

How about `GVarPod_APS<int> app_num_workers` - when 0, auto-detected and updated (left as 0 in RoR.cfg).

**Comment by ulteq** [comment]
That variable would then be used for changing the amount of workers on the fly?

**Comment by ulteq** [comment]
I should have used `GVarPod_APS` as you suggested. :man_facepalming:

**Comment by ulteq** [comment]
Ready, from my point of view.

**Comment by ulteq** [comment]
Ready, from my point of view.

**Comment by ulteq** [comment]
@only-a-ptr anything you want to add?

File: source/main/physics/BeamFactory.cpp [hunk 1]
```diff
@@ -1023,18 +1008,14 @@ void ActorManager::UpdateActors(Actor* player_actor, float dt)
...
-    {
-        this->UpdatePhysicsSimulation();
-    }
+    auto func = std::function<void()>([this]()
+        {
+            this->UpdatePhysicsSimulation();
+        });
+    m_sim_task = gEnv->threadPool->RunTask(func);
+
+    if (!RoR::App::app_async_physics.GetActive())
```
Comment (only-a-ptr) [inline source/main/physics/BeamFactory.cpp]: This reminds me of the "AsynchronousPhysics" setting in 0.4.0.7 which I removed as broken by design. Unless I'm missing something, it should cause a race condition under heavier physics workload. `join()`ing the sim. tasks here guards you against overlapping physics updates - without it, you may start another round of updates before the previous one finishes, eventually causing the same actor (nodes, beams, engine...) to be modified by 2 threads, the current step and the previous step. Correct me if I'm wrong.

**Comment by only-a-ptr** [comment]
I didn't find the time to test, so I just skimmed over the code. I'm confident it will run just fine.

**Comment by only-a-ptr** [comment]
I wanted to have a thorough look on this, but finally I didn't find the time, so I just skimmed over the code now. I'm confident it will run just fine.

**Pull request merged**